### PR TITLE
fix: `HaiProxy` doesn't revert if the target has no code

### PR DESCRIPTION
**Pull request by 0xJabberwock**
Closes HAI-278

### CODE REVIEW

**Comment by xBA5ED** [comment]
Blocked by #8, once its merged we can use custom errors

File: src/contracts/proxies/HaiProxy.sol
Comment (0xJabberwock) [inline src/contracts/proxies/HaiProxy.sol:30]: This check isn't really required as OpenZeppelin's `Address` library would revert as well with an `address(0)` argument (since no code can be found there).

**Review by wei3erHase — APPROVED** [review]
great!

**Pull request merged**
